subprojects/docs/src/docs/userguide/kotlin_dsl.adoc

```diff
@@ -603,165 +483,215 @@ include::sample[dir="kotlinDsl/containers/scope",files="build.gradle.kts[tags=sc
+<4> Does the same as the previous line but allows the property to have a null value
 
-Let's say the snippet above is from your root project build script, in order to reference the `myNewProperty` from a sub project you'd do:
+This approach works for all Gradle scripts: project build scripts, script plugins, settings scripts and initialization scripts.
+
+You can also access extra properties on a root project from a subproject using the following syntax:
 
 .my-sub-project/build.gradle.kts
 [source,kotlin]
 ----
-val myNewProperty: String by rootProject.extra
+val myNewProperty: String by rootProject.extra  // <1>
 ----
+<1> Binds the root project's `myNewProperty` extra property to a reference of the same name
 
-It works the same way in settings or init scripts.
-
-All `ExtensionAware` types can host extra properties, `Task` is a common example:
+Extra properties aren't just limited to projects.
+For example, `Task` extends `ExtensionAware`, so you can attach extra properties to tasks as well.
+Here's an example that defines a new `myNewTaskProperty` on the `test` task and then uses that property to initialize another task:
 
 .build.gradle.kts
 [source,kotlin]
 ----
 tasks {
     test {
-        // Define new extra properties
-        val myNewTaskProperty by extra("initial value")
-        val myOtherNewTaskProperty by extra { "lazy initial value" }
+        val reportType by extra("dev")  // <1>
+        doLast {
+            // Use 'suffix' for post processing of reports
+        }
+    }
 
-        // Reference existing extra properties
-        val myTaskProperty: String by extra
-        val myNullableTaskProperty: String? by extra
+    register<Zip>("archiveTestReports") {
+        val reportType: String by test.get().extra  // <2>
+        appendix = reportType
+        from(test.get().reports.html.destination)
     }
 }
 ----
+<1> Creates a new `reportType` extra property on the `test` task
+<2> Makes the `test` task's `reportType` extra property available to configure the `archiveTestReports` task
+
+If you're happy to use eager configuration rather than the configuration avoidance APIs, you could use a single, "global" property for the report type, like this:
+
+.build.gradle.kts
+[source,kotlin]
+----
+tasks.test.doLast { ... }
+
+val testReportType by tasks.test.extra("dev")  // <1>
+
+tasks.create<Zip>("archiveTestReports") {
+    appendix = testReportType  // <2>
+    from(test.get().reports.html.destination)
+}
+----
+<1> Creates and initializes an extra property on the `test` task, binding it to a "global" property
+<2> Uses the "global" property to initialize the `archiveTestReports` task
+
+There is one last syntax for extra properties that we should cover, one that treats `extra` as a map.
+We recommend against using this in general as it bypasses Kotlin's type checking and prevents IDEs from providing as much support as they could.
```

Comment:
:x: it doesn't bypass type checking, it exposes extra properties as `Any?`

Reply:
And that doesn't bypass type checking? If all types were `Object`, I'd consider that bypassing type checking. How would you rephrase it, because you're definitely losing out on the benefits of static type checking whether it's being "bypassed" or not.

Reply:
yes, you lose type checking because you interact with `Any?`
but I would not use the word "bypass"